### PR TITLE
Add mkdocs scaffolding

### LINKED ISSUE #1359: Create mkdocs scaffolding to build the docs locally
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx-operator/issues) for duplicates.
- [X] I understand that AWX Operator is open source software provided for free and that I might not receive a timely response.

### Feature Summary

Based on[ MkDocs](https://www.mkdocs.org/getting-started/), create the scaffolding required to build the docs locally.

Use https://github.com/ansible/ansible-navigator/blob/main/mkdocs.yml as an example of how to do this.

NOTE - we may not need everything in that example but it's a good working project to learn from

### PR DESCRIPTION
**Pull request by BhattacharjeeSutapa**
##### SUMMARY

Adding mkdocs scaffolding 

Fixes #1359 

##### ISSUE TYPE

 - New or Enhanced Feature 

So far I am hitting the following error when I try to build the docs locally. 

```
$ mkdocs serve
INFO     -  Building documentation...
INFO     -  Cleaning site directory
INFO     -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
              - debugging.md
              - migration.md
WARNING  -  A relative path to 'index.md' is included in the 'nav' configuration, which is not found in the documentation files
INFO     -  Documentation built in 5.62 seconds
INFO     -  [23:55:54] Watching paths for changes: 'docs', 'mkdocs.yml'
INFO     -  [23:55:54] Serving on http://127.0.0.1:8000/
WARNING  -  [23:56:02] "GET / HTTP/1.1" code 404

```
Due to this error, when I open the page, it throws a 404 error. Any help is appreciated. For this reason, I have `strict: false` so that the site at least serves.

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
+---
+site_name: awx-operator
+site_url: https://awx-operator.readthedocs.io/
+repo_url: https://github.com/ansible/awx
```
Comment (samccann) [inline mkdocs.yml]:
```suggestion
repo_url: https://github.com/ansible/awx-operator
```

**Comment by samccann** [comment]
To fix the warning, can you:
1. create docs/index.md
2. add the two existing files to that?

That should build a simple docsite with those two existing files.

**Comment by samccann** [comment]
Okay looks like we can potentially get away without an index.md file if we follow what was done here -https://github.com/ansible/ansible-language-server/blob/main/mkdocs.yml

**Comment by samccann** [comment]
@BhattacharjeeSutapa - the following as a requirements.txt file in the docs folder might help avoid having to do an apt install mkdocs...

cairosvg
markdown-exec>=1.0.0
mkdocs-gen-files>=0.4.0
mkdocs-material-extensions>=1.1.1
mkdocs-material>=9.0.6
mkdocs>=1.4.2
mkdocstrings-python>=0.8.3
mkdocstrings>=0.20.0
pillow
pipdeptree>=2.3.3
pymdown-extensions>=9.9.2

Then a pip install -r requirements.txt might work instead of an apt install

File: mkdocs.yml (hunk 1 above)
Comment (BhattacharjeeSutapa) [reply mkdocs.yml]: This is now done.

**Comment by samccann** [comment]
@oraNod - can you review this PR when you have the chance?

File: mkdocs.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
+---
+site_name: awx-operator
+site_url: https://awx-operator.readthedocs.io/
+repo_url: https://github.com/ansible/awx-operator
+edit_uri: blob/main/docs/
```
Comment (oraNod) [inline mkdocs.yml]:
```suggestion
edit_uri: blob/devel/docs/
```

**Review by oraNod — COMMENTED** [review]
Nice work @BhattacharjeeSutapa I can build the site without error. One nit on the uri that allows direct edits on GitHub.

Should this PR include a title and maybe some starter text in "index.md"?

**Comment by oraNod** [comment]
Looking good @BhattacharjeeSutapa One final nit is to move that requirements file into the docs directory (`docs/requirements.txt`). That keeps it separate so you don't wind up installing unneeded packages if you aren't building the docs. Thank you for working so diligently on this!

**Comment by BhattacharjeeSutapa** [comment]
> Looking good @BhattacharjeeSutapa One final nit is to move that requirements file into the docs directory (`docs/requirements.txt`). That keeps it separate so you don't wind up installing unneeded packages if you aren't building the docs. Thank you for working so diligently on this!

Moved requirements.txt file

**Review by oraNod — APPROVED** [review]
Thank you @BhattacharjeeSutapa

**Review by samccann — APPROVED** [review]

**Comment by samccann** [comment]
@BhattacharjeeSutapa hey I may not be online at the usual time tomorrow but looks like this PR is failing automated CI testing. In case it doesn't show the error, this is what I'm seeing:
image

So basically it's saying on that line in the mkdocs.yml file, there is an extra whitespace that you need to delete.

Thanks!

**Comment by oraNod** [comment]
@TheRealHaoLiu Hi, This PR adds docs scaffolding to build AWX Operator docs with MKDocs and is failing CI with an error related to molecule. Would you mind taking a look at that please? Not sure how to fix that. Thanks.

**Comment by BhattacharjeeSutapa** [comment]
> @BhattacharjeeSutapa hey I may not be online at the usual time tomorrow but looks like this PR is failing automated CI testing. In case it doesn't show the error, this is what I'm seeing: image
> 
> So basically it's saying on that line in the mkdocs.yml file, there is an extra whitespace that you need to delete.
> 
> Thanks!

Hi @samccann two whitespaces have been deleted. Sorry for late update.

**Comment by samccann** [comment]
@BhattacharjeeSutapa - hey - it took me so long to get back to this that your branch now needs a rebase.  If you're unfamiliar with rebasing in git, you can see the basics at https://docs.github.com/en/get-started/using-git/about-git-rebase  and ping any of us on the Ansible docs matrix channel if you want more assistance.  Note the main branch here is called devel so you'll be rebasing to that.

For me, the typical steps are:
git checkout devel
git fetch <origin> devel
git checkout mybranch
git rebase devel

Hopefully you don't get any merge conflicts but if you do, we can help you resolve them in matrix  or here if you prefer.
If there are no merge conflicts, the final step is:
git push mybranch -f

**Comment by samccann** [comment]
@BhattacharjeeSutapa - along with the rebase, can you also add the /site folder to the .gitignore? This will make sure git doesn't try to add the temporary /site files to anyone's PR after they do a mkdocs build.

**Comment by oraNod** [comment]
@BhattacharjeeSutapa Hi, there are two things needed to pass the Molecule CI tests. First thing is to lint the mkdoc.yaml file. It has a couple of undefined fields and a whitespace issue.

I made the changes in this commit: https://github.com/oraNod/awx-operator/commit/3133f246849cbd9310d40910fa728b8861c1a016

You can cherry-pick that or do your own linting in something like VSCode (ctrl+i).

The 2nd thing is that the docs requirements need to be updated. I've made the changes in this commit, feel free to cherry-pick them: https://github.com/oraNod/awx-operator/commit/399a23516255123f5442c9f2f02b3b7d36472b25

After that the Molecule tests should pass (I tested with a PR that includes your commits plus my two: https://github.com/ansible/awx-operator/actions/runs/5387656775/jobs/9779262790?pr=1469)

Hope this helps!

**Comment by samccann** [comment]
thanks @BhattacharjeeSutapa for working with us to get  this merged!

**Pull request merged**